flatpak: 1.0.5 -> 1.1.2

dtzWill opened the pull request:
###### Motivation for this change

WIP-ish, at least it builds now :).


IIRC there are compelling reasons to upgrade
(UI and maybe sandbox improvements) but will
need to look through release notes again.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

dtzWill committed:
flatpak: 1.0.5 -> 1.1.2

dtzWill commented:
@GrahamcOfBorg build flatpak
@GrahamcOfBorg test flatpak

jtojnar commented:
This is a dev version but other distros package it as well https://repology.org/metapackage/flatpak/versions so perhaps it is fine.

We should also update xdg-desktop-portal and xdg-desktop-portal-gtk.

dtzWill committed:
xdg-desktop-portal: 1.0.3 -> 1.1.0

Do not add geoclue dependency yet:

Complains it's not new enough, apparently wants
an untagged version (2.5.1 is latest tag)

checking for libgeoclue-2.0 >= 2.5.2... no
configure: error: Package requirements (libgeoclue-2.0 >= 2.5.2) were
not met:

Requested 'libgeoclue-2.0 >= 2.5.2' but version of Geoclue Client
Library is 2.5.1

dtzWill committed:
xdg-desktop-portal-gtk: 1.0.2 -> 1.1.0

Lockdown requires gsettings-desktop-schemas

jtojnar commented on pkgs/development/libraries/xdg-desktop-portal/default.nix:
```diff
@@ -11,20 +11,22 @@ in stdenv.mkDerivation rec {
 
   patches = [
     ./respect-path-env-var.patch
+    ./0001-test-doc-portal-Skip-all-tests-if-FUSE-isn-t-support.patch
```
Please add a link to the source.

jtojnar commented on pkgs/development/libraries/xdg-desktop-portal-gtk/default.nix:
```diff
@@ -1,15 +1,15 @@
 let
-  version = "1.0.2";
+  version = "1.1.0";
 in stdenv.mkDerivation rec {
```
The lockdown will require `gsettings-desktop-schemas`.

https://github.com/flatpak/xdg-desktop-portal-gtk/blob/14c62ab18b49e67324b9a75e3627b075c5a6067d/src/lockdown.c#L48

For an example how to do it see https://github.com/NixOS/nixpkgs/pull/42177/files#diff-81007e7b56315f92406c1cf3d7ee1d47R16

dtzWill replied:
Is tackling the lockdown aspect a blocker? I can look into a solution similar to what you've linked, but might take a bit (may need to start testing using a proper gnome session so I can access these things?).

On the other hand if the "fail" mode here isn't really "fail-safe" (such that user requests permission be denied but it's ignored since flatpak can't find the setting or something) that's no good and certainly should be addressed!

FWIW upstream has more changes re:lockdown (https://github.com/flatpak/xdg-desktop-portal/pull/267), not sure if that makes integration easier or not :).

jtojnar replied:
If schemas are not available, the program will crash.

jtojnar commented:
When running `nix-build -A tests.flatpak.x86_64-linux nixos/release.nix`, many tests fail with _Floating point exception_:

```
FAIL: flatpak/test-info@system.wrap.test (Child process exited with code 136)
FAIL: flatpak/test-run@system,deltas.wrap.test (Child process exited with code 136)
FAIL: flatpak/test-repo@collections.wrap.test (Child process exited with code 136)
FAIL: flatpak/test-oci-registry@user.wrap.test (Child process exited with code 136)
FAIL: flatpak/test-unsigned-summaries.sh.test (Child process exited with code 136)
FAIL: flatpak/test-oci-registry@system.wrap.test (Child process exited with code 136)
FAIL: flatpak/test-build-update-repo.sh.test (Child process exited with code 136)
FAIL: flatpak/test-run@system,nodeltas.wrap.test (Child process exited with code 136)
FAIL: flatpak/test-run@user,nodeltas.wrap.test (Child process exited with code 136)
FAIL: flatpak/test-run@user,deltas.wrap.test (Child process exited with code 136)
FAIL: flatpak/test-update-remote-configuration.sh.test (Child process exited with code 136)
FAIL: flatpak/test-info@user.wrap.test (Child process exited with code 136)
FAIL: flatpak/test-override.sh.test (Child process exited with code 136)
FAIL: flatpak/test-extensions.sh.test (Child process exited with code 1)
FAIL: flatpak/test-bundle@system.wrap.test (Child process exited with code 136)
FAIL: flatpak/test-repo@user.wrap.test (Child process exited with code 136)
FAIL: flatpak/test-repo@system.wrap.test (Child process exited with code 136)
FAIL: flatpak/test-repo@collections-server-only.wrap.test (Child process exited with code 136)
FAIL: flatpak/test-bundle@user.wrap.test (Child process exited with code 136)
```

```
Running test: flatpak/test-info@system.wrap.test
1..7
78293 metadata, 3735 content objects imported                                     
runtime/org.test.Platform/x86_64/master: 8592b48c2c50b4d5a146b04e7187a2b9f5ba3548811f28ba2444d6e886046a87
Updating appstream branch
Extracting icons for component org.test.Hello
Updating summary
Running web server: PYTHONUNBUFFERED=1 setsid python -m SimpleHTTPServer 0
Web server pid: 5313
Waiting for web server (1/300)...

Waiting for web server (2/300)...

Waiting for web server (3/300)...

Waiting for web server (4/300)...

Waiting for web server (5/300)...

Waiting for web server (6/300)...

Started web server 'python -m SimpleHTTPServer 0': process 5313 on port 38723
127.0.0.1 - - [02/Jan/2019 23:59:59] "GET /test/summary.sig HTTP/1.1" 200 -
127.0.0.1 - - [02/Jan/2019 23:59:59] "GET /test/summary HTTP/1.1" 200 -
Looking for matches…
127.0.0.1 - - [02/Jan/2019 23:59:59] "GET /test/summary.sig HTTP/1.1" 200 -
127.0.0.1 - - [02/Jan/2019 23:59:59] "GET /test/summary.sig HTTP/1.1" 200 -


/nix/store/x5nj18mi7rsxqylqhsyvsmg20grg31rm-flatpak-1.1.2-installedTests/libexec/installed-tests/flatpak/libtest.sh: line 304:  5384 Floating point exception${FLATPAK} ${U} install -y ${REPONAME}-repo org.test.Platform master
OK closing connection
fusermount: failed to unmount /tmp/test-flatpak-CEqpVk/runtime/doc: No such file or directory
FAIL: flatpak/test-info@system.wrap.test (Child process exited with code 136)
```

That does not seem to happen when running the tests outside of VM using `env XDG_DATA_DIRS=(nix-build -A flatpak.installedTests)/share nix-shell -p gnome-desktop-testing --run 'gnome-desktop-testing-runner flatpak/test-info@system.wrap.test'`

jtojnar commented:
I tested successfully installing flatpak app and running it.

dtzWill commented:
Yay!! Thanks for your work on this!

On Thu, 03 Jan 2019 16:23:20 +0000 (UTC), Jan Tojnar <notifications@github.com> wrote:
> Merged #52754 into master.
> 
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/NixOS/nixpkgs/pull/52754#event-2052748408Non-text part: text/html
